core/shared/src/main/scala/fs2/compression/GunzipResult.scala

```diff
@@ -0,0 +1,159 @@
+  * @param comment File comment.
+  */
+class GunzipResult[F[_]](
+    val modificationTimeEpoch: Option[FiniteDuration],
+    val fileName: Option[String],
+    val comment: Option[String],
+    val content: Stream[F, Byte]
+) extends Product
+    with Serializable
+    with Equals {
+
+  def modificationTime: Option[Instant] =
+    modificationTimeEpoch.map(e => Instant.ofEpochSecond(e.toSeconds))
+
+  @deprecated(
+    "use the default constructor with modificationTimeEpoch: Option[FiniteDuration]",
+    "x.x"
+  )
+  def this(
+      content: Stream[F, Byte],
+      modificationTime: Option[Instant] = None,
+      fileName: Option[String] = None,
+      comment: Option[String] = None
+  ) = this(
+    modificationTime.map(i => FiniteDuration(i.getEpochSecond, TimeUnit.SECONDS)),
+    fileName,
+    comment,
+    content
+  )
+
+  @deprecated("GunzipResult is no longer a case class.", "x.x")
+  def copy(
+      content: Stream[F, Byte] = this.content,
+      modificationTime: Option[Instant] = this.modificationTime,
+      fileName: Option[String] = this.fileName,
+      comment: Option[String] = this.comment
+  ): GunzipResult[F] = new GunzipResult[F](
+    content,
+    modificationTime,
+    fileName,
+    comment
+  )
+
+  @deprecated("GunzipResult is no longer a case class.", "x.x")
+  def productArity: Int = 4
+
+  @deprecated("GunzipResult is no longer a case class.", "x.x")
+  def productElement(n: Int): Any = n match {
+    case 0 => content
+    case 1 => modificationTimeEpoch
+    case 2 => fileName
+    case 3 => comment
+  }
+
+  def canEqual(other: Any): Boolean = other match {
+    case _: GunzipResult[_] => true
+    case _                  => false
+  }
+
+  override def equals(other: Any): Boolean = other match {
+    case that: GunzipResult[_] =>
+      (that.canEqual(this)) &&
+        modificationTimeEpoch == that.modificationTimeEpoch &&
+        fileName == that.fileName &&
+        comment == that.comment &&
+        content == that.content
+    case _ => false
+  }
+
+  override def hashCode(): Int = {
+    val state = Seq(content, modificationTimeEpoch, fileName, comment)
+    state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
+  }
+
+  override def toString = s"GunzipResult($content, $modificationTimeEpoch, $fileName, $comment)"
+
+}
+
+object GunzipResult {
+
+  @deprecated("Use GunzipResult.create", "x.x")
+  def apply[F[_]](
+      content: Stream[F, Byte],
+      modificationTime: Option[Instant] = None,
+      fileName: Option[String] = None,
+      comment: Option[String] = None
+  ): GunzipResult[F] = new GunzipResult(
+    content,
+    modificationTime,
+    fileName,
+    comment
+  )
+
+  def create[F[_]](
+      content: Stream[F, Byte],
+      modificationTimeEpoch: Option[FiniteDuration] = None,
+      fileName: Option[String] = None,
+      comment: Option[String] = None
+  ): GunzipResult[F] = new GunzipResult(
+    modificationTimeEpoch,
+    fileName,
+    comment,
+    content
+  )
```

Comment:
I may be missing something, but I'm actually not sure if we need a public constructor for this class. Seems like it would only be created inside of fs2.

Reply:
Made it private.